### PR TITLE
Allow dots in keywords when searching on exact words

### PR DESCRIPTION
**Pull request by stellarhoof**
Allow dots in keywords tags when matching on exact words

https://github.com/smartprocure/contexture/assets/4336260/c44b7dc6-453a-4d60-9b6d-781f068906ec

### CODE REVIEW

File: packages/react/src/exampleTypes/ExpandableTagsQuery/index.js
Comment (conraddavisjr) [inline packages/react/src/exampleTypes/ExpandableTagsQuery/index.js]:
```suggestion
    // Allow tag keywords to contain dots in them if the user is searching for exact
```

**Review by conraddavisjr — APPROVED** [review]
LGTM! Pending one more ✅ from another reviewer.
I dropped feedback for one minor typo adjustment

File: packages/react/src/exampleTypes/ExpandableTagsQuery/index.js
Comment (dubiousdavid) [inline packages/react/src/exampleTypes/ExpandableTagsQuery/index.js]: Why would we not always use `alphaNumericRegExWithDots`? Also, why are the regexs so strange looking?

Comment (stellarhoof) [reply packages/react/src/exampleTypes/ExpandableTagsQuery/index.js]: I thought that's we had agreed on, that if you didn't search for exact you'd get the current behavior?

I think they are strange looking because they were copied from lodash. See [this comment](https://github.com/smartprocure/contexture/pull/203/files#diff-431701c69ca84ff7bf09ad61f1da89a10d000c32116db6b3795354611d40de6eR40)

**Pull request merged**
